src/pymoca/parser.py

```diff
@@ -810,3 +826,262 @@ def parse(text: str) -> Union[ast.Tree, None]:
     parse_walker.walk(ast_listener, parse_tree)
     modelica_file = ast_listener.file_node
     return file_to_tree(modelica_file)
+
+
+def _microseconds_since_epoch(d: Optional[datetime] = None) -> int:
+    if d is None:
+        d = datetime.utcnow()
+    return int(d.timestamp() * 1e6)
+
+
+def _check_database_structure(conn):
```

Comment:
> or because we don't want to type hint non-API code?
It was subconsciously that I didn't (not a matter of "wanting"), but I think it indeed is public API that I tend to type. I'm totally not opposed to typing the internal/private API though, I think we actually ought to do that for readability for ourselves.